UNITTESTS/README.md

```diff
@@ -0,0 +1,357 @@
+
+Debian/Ubuntu/Mac OS:
+```
+mkdir UNITTESTS/build
+cd UNITTESTS/build
+cmake -DCMAKE_BUILD_TYPE=Debug -DCOVERAGE:STRING=html  ..
+make
+./features_netsocket_InternetSocket
+gcovr -r ../.. --html --html-detail -o ./index.html ./CMakeFiles/features_netsocket_InternetSocket.MbedOS.dir/
+```
+Windows:
+```
+mkdir UNITTESTS/build
+cd UNITTESTS/build
+cmake -DCMAKE_BUILD_TYPE=Debug -DCOVERAGE:STRING=html -g "MinGW Makefiles" ..
+mingw32-make
+features_netsocket_InternetSocket.exe
+gcovr -r ..\.. --html --html-detail -o .\index.html .\CMakeFiles\features_netsocket_InternetSocket.MbedOS.dir\
+```
+
+## The structure of unit tests
+
+The structure of the unit tests directory looks like this:
+```
+UNITTESTS
+  ├── mbed_unittest.py                       Python tool for unit testing
+  ├── unit_test                              Python tool modules
+  ├── CMakeLists.txt                         CMake project definition file
+  ├── CMakeSettings.json                     CMake configurations for Visual Studio 2017
+  ├── README.md
+  ├── googletest-CMakeLists.txt.in           CMake project definition file for Google Test
+  │
+  ├── features
+  │   └── netsocket                          Directory tree that mirrors Mbed OS root
+  │       ├── NetworkInterface               Name of the class to be tested
+  │       │   ├── test_NetworkInterface.cpp
+  │       │   └── unittest.cmake             CMake module for unit test
+  │       └── Socket
+  │
+  ├── stubs                                  Shared stubs which can be used for tests.
+  ├── target_h                               Shared headers which can be used for tests.
+  └── template                               Templates for creating new unittests
+```
+Each unit test has an identical directory tree as seen in the Mbed OS root folder. This is not a mandatory requirement but helps to maintain tests. Each class to be tested have their own `unittest.cmake` which is found by `CMakeLists.txt`.
+
+## Creating a unit test
+
+Each class to be tested requires two files for unit testing:
+1. C++ unit test source file (e.g. `test_NetworkInterface.cpp`)
+2. CMake module file for unit test definition (`unittest.cmake`)
+
+A unit test definition file `unittest.cmake` requires variables to be set for a test to be configured. File source paths in `unittest.cmake` files need to be relative to the unit test folder and `CMakeLists.txt`.
+
+* **TEST_SUITE_NAME** - Identifier for the test suite. Use naming convention *PATH_TO_THE_TESTABLE_FILE* e.g. *features_netsocket_InternetSocket*
+* **unittest-includes** - Include paths for headers needed to build the tests in addition to the base include paths listed in [CMakeLists.txt](CMakeLists.txt). Optional.
+* **unittest-sources** - Mbed OS source files and stubs included for the build.
+* **unittest-test-sources** - Unit test source files.
+
+#### Creating unit tests files with Mbed CLI
+
+```
+mbed unittest new
+```
+
+##### Parameters to mbed unittest new
```

Comment:
Don't list parameters here because there is no guarantee. Just command and why it exist is good enough